### DISCUSSION TITLE
How to make so that each paragraph has a unique-ish ID?

### DISCUSSION THREAD

**Discussion by euvl** [ueberdosis/tiptap#1421] [General Questions]
Hi guys, 

I am trying to understand how to make so that each paragraph would have a unique ID?

I am planning to have "comments" stored against each paragraph. My thinking was that I will need an ID of a paragraph that is unique (within the context of a single document) and persistent (survives serialization & storage in DB).

Does anyone have any idea if/how this could be done? 🙏

**Comment by philippkuehn** [comment]
good news: I’m already working on an unique id extension :)

**Reply by euvl** [reply]
oh, amazing!

**Reply by arnogramatke** [reply]
Are there any news on this? I would like to use unique ids to be able to compute changes between different versions of the same document.

**Reply by matkinjuliectrl** [reply]
I think this is it... seems like the only 'pro' extension there is though so you have to pay for it https://tiptap.dev/api/extensions/unique-id
